src/domain/mainwindow.py

```diff
@@ -358,10 +355,16 @@ def configure_email(self):
             return
 
     def check_notification_enablement(self):
-        """Method to check whether a notification protocol has been set in WADAS."""
-
-        credentials = keyring.get_credential("WADAS_email", "")
-        if not self.email_config["smtp_hostname"] or not credentials.username:
+        """Method to check whether a notification protocol has been set in WADAS.
+        If not, ask the user whether to proceed without."""
+
+        notification = False
+        for notifier in notifiers:
+            if notifier.type == Notifier.NotifierTypes.Email:
+                credentials = keyring.get_credential("WADAS_email", "")
```

Comment:
This way you pick the last notifier found: is it what do you expect?

Reply:
The idea is to have at least one notifier configured... As of today only email is supported, in future in the notifier.type check we will add other notifier and related logic.

Reply:
Then maybe it's better to have a class hierarchy for the notifiers, that will be called when needed, instead of checking their enum value with ifs spread around the code.
Each notifier will subclass the super class and implement its specific business logic.

Reply:
It is already a hierarchy of classes. Notifier -> EmailNotifier. As multiple types notifier can be enabled simoultaneously (eg. email, whazzup, etc) here I simply check how many types are enabled and prepare to trigger notifications.

Reply:
Then you can introduce a method in the class hierarchy where to incapsulate the code for each different subclass, instead of checking the notifier type and then execute the code.